src/libs/actions/IOU.ts

```diff
@@ -7900,6 +7900,33 @@ function putOnHold(transactionID: string, comment: string, reportID: string) {
         },
     ];
 
+    if (hash) {
```

Comment:
We could do with an explanatory comment here, like `If we are holding from the search page, we optimistically update the snapshot data that search uses so that it is kept in sync`.